java-connectors/kafka-connect-common/src/main/java/io/lenses/streamreactor/common/util/JarManifest.java

```diff
@@ -69,11 +73,13 @@ public JarManifest(JarFile jarFile) {
     Manifest manifest;
     try {
       Optional<JarFile> jarFileOptional = of(jarFile);
-      manifest = jarFileOptional.get().getManifest();
+      if (jarFileOptional.isPresent()) {
+        manifest = jarFileOptional.get().getManifest();
```

Comment:
It would be nice to avoid the `.get` on the Optional here.

Reply:
+1 on working with Optionl

```java
of(jarFile).flatMap(j-> of(j.getManifest)).flatMap(m->of(m.getMainAttributes).map(extractMainAttributes).orNull
```

Reply:
Unfortunately it's not that easy as `getManifest()` can give us `IOException` which we probably wouldn't like to try to catch inside of functional stream. Still, I tried to make it as functional as possible, which I hope will be a nice consensus (it'll become a monad but there will be no try-catch inside) between our ideas.